src/Diagnostics.RuntimeHost/Services/ResourceService/SiteService.cs

```diff
@@ -26,13 +27,21 @@ public async Task<StackType> GetApplicationStack(string subscriptionId, string r
             if (string.IsNullOrWhiteSpace(siteName)) throw new ArgumentNullException("siteName");
 
             string queryTemplate =
-                $@"WawsAn_dailyentity 
+                $@"WawsAn_dailyentity
                 | where pdate >= ago(5d) and sitename =~ ""{siteName}"" and sitesubscription =~ ""{subscriptionId}"" and resourcegroup =~ ""{resourceGroup}"" 
                 | where sitestack !contains ""unknown"" and sitestack !contains ""no traffic"" and sitestack  !contains ""undefined""
                 | top 1 by pdate desc
                 | project sitestack";
 
-            DataTable stackTable = await dp.Kusto.ExecuteQuery(queryTemplate, DataProviderConstants.FakeStampForAnalyticsCluster, operationName: "GetApplicationStack");
+            DataTable stackTable = null;
+            
+            try{
+                stackTable = await dp.Kusto.ExecuteQuery(queryTemplate, DataProviderConstants.FakeStampForAnalyticsCluster, operationName: "GetApplicationStack");
+            }catch(Exception ex){
+                //swallow the exception. Since Mooncake does not have an analytics cluster
+                DiagnosticsETWProvider.Instance.LogRuntimeHostHandledException(dataProviderContext.RequestId, "GetApplicationStack", subscriptionId,
```

Comment:
Is there a way to know which cloud we are running in? like Public or Non-Public.
There is a dependency of stack in Filters as well as we pass the data into Context for detector authors to use.

Another orthogonal question - what if there is an exception here. What is getting Passed to context? Is it StackType.None

Reply:
Yep StackType.None would be in the context. There might be a way to find out what cloud we're in but I didn't pursue that route as I think this is the fastest route.